### PR TITLE
Fix E2E tests

### PR DESCRIPTION
**Pull request by gigitux**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Currently, we run E2E test suite every day against:
- latest version of WordPress
- L-1 version of WordPress
- Nightly version of WordPress

We need to ensure that E2E tests don't fail against those.  This PR improves compatibility of the E2E test suite with previous WordPress version.

Closes # .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure that E2E tests are green in this PR.
2. Ensure that E2E tests [in this PR](https://github.com/woocommerce/woocommerce/pull/52932) are green (it is the same PR, the only change is the WordPress version)

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [x] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 
Fix E2E test suite to support WordPress 6.6

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/tests/e2e/tests/templates/template-customization.block_theme.spec.ts [hunk 1]
```diff
@@ -41,11 +42,16 @@ test.describe( 'Template customization', () => {
 				await editor.saveSiteEditorEntities( {
 					isOnlyCurrentEntityDirty: true,
 				} );
+
+				const name =
```
Comment (adimoldovan) [inline plugins/woocommerce-blocks/tests/e2e/tests/templates/template-customization.block_theme.spec.ts]: The name option in getByRole supports Regex. I think you should be able to express this locator without needing the WordPress version.

File: plugins/woocommerce-blocks/tests/e2e/utils/test.ts [hunk 2]
```diff
@@ -168,6 +173,26 @@ const test = base.extend<
...
+		}
+
+		if ( installedVersion === stableVersion ) {
+			await use( 'stable' );
+		}
+
+		if ( installedVersion === nightlyVersion ) {
+			await use( 'nightly' );
+		}
+	},
```
Comment (adimoldovan) [inline plugins/woocommerce-blocks/tests/e2e/utils/test.ts]: I think you're missing a default case when the version is none of the 3 defined.

**Review by adimoldovan — COMMENTED** [review]
Thanks for working on this @gigitux.
I'm not entirely sure it will work, but I think you can rely on RegEx and drop the whole WordPress version check.
https://playwright.dev/docs/api/class-locator#locator-get-by-role-option-name

**Comment by gigitux** [comment]
> Thanks for working on this @gigitux. I'm not entirely sure it will work, but I think you can rely on RegEx and drop the whole WordPress version check. https://playwright.dev/docs/api/class-locator#locator-get-by-role-option-name

Do you think that we should avoid relying on the WordPress version? Even it could not be necessary now, I think that it could be useful in the future too.

**Comment by adimoldovan** [comment]
> > Thanks for working on this @gigitux. I'm not entirely sure it will work, but I think you can rely on RegEx and drop the whole WordPress version check. [playwright.dev/docs/api/class-locator#locator-get-by-role-option-name](https://playwright.dev/docs/api/class-locator#locator-get-by-role-option-name)
> 
> Do you think that we should avoid relying on the WordPress version? Even it could not be necessary now, I think that it could be useful in the future too.

Given that we're only supporting L-1 version I would avoid it if possible.

**Comment by gigitux** [comment]
@adimoldovan I updated the PR! Thanks for the review!

**Review by adimoldovan — APPROVED** [review]
@gigitux Nice! The updates look good to me. I find it much cleaner now without the need to handle WP version. Thank you for this!

**Pull request merged**